__tests__/app.test.js

```diff
@@ -191,3 +191,104 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Inserts a new comment to the db and sends the comment back to the client", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "New comment",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment).toHaveProperty("comment_id");
+        expect(comment).toHaveProperty("votes");
+        expect(comment).toHaveProperty("created_at");
+        expect(comment).toHaveProperty("author");
+        expect(comment).toHaveProperty("body");
+        expect(comment).toHaveProperty("article_id");
+      });
+  });
+
+  test("400: Responds with an bad request error message when an non-existing username is provided", () => {
+    const newComment = {
+      username: "wrong_username",
+      body: "New comment",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("bad request");
+      });
+  });
```

Comment:
Nice error handling test, but in this case this should be a 404. The reason for this is `"wrong_username"` is simply a username that does not exist but it is still a valid username and could be someones username in the future, so this should be a 404 as the user does not exist.